supertokens_python/recipe/emailverification/api/implementation.py

```diff
@@ -23,19 +23,39 @@
 
 if TYPE_CHECKING:
     from supertokens_python.recipe.emailverification.interfaces import (
-        APIOptions, GenerateEmailVerifyTokenPostResponse, IsEmailVerifiedGetResponse, EmailVerifyPostResponse
+        APIOptions, GenerateEmailVerifyTokenPostResponse, IsEmailVerifiedGetResponse
     )
 
 from supertokens_python.recipe.emailverification.types import User
 from supertokens_python.recipe.session.asyncio import get_session
 
 
 class APIImplementation(APIInterface):
-    async def email_verify_post(self, token: str, api_options: APIOptions, user_context: Dict[str, Any]) -> EmailVerifyPostResponse:
+    async def email_verify_post(self, token: str, api_options: APIOptions, user_context: Dict[str, Any]) -> Union[EmailVerifyPostOkResponse, EmailVerifyPostInvalidTokenErrorResponse]:
+        """email_verify_post is an async mehtod that accepts
+
+               - token: string,
+               - api_options: APIOptions,
+               - user_context: python dict
+
+            if the user is successfully varified, the response could be as follows
+
+                {
+                    "status": "OK",
+                    "user": {
+                      "id": "<usern-id>",
+                      "email": "<email-id>"
+                }
+
+            If the token is invalid and the user is not varified,
+            the response could be as follows:
+
+                {
+                    "status": "EMAIL_VERIFICATION_INVALID_TOKEN_ERROR"
+                }
+        """
         response = await api_options.recipe_implementation.verify_email_using_token(token, user_context)
-        if response.is_ok:
-            if response.user is None:
-                raise Exception("Should never come here")
+        if response.user:
```

Comment:
This code should not change.

Reply:
Thanks for the review @rishabhpoddar .

I think we should change this.

At this point we always have the response which is an instance of `VerifyEmailUsingTokenResult`.
and we are ensuring that the user is a valid object of the class User before sending the response object which is
`EmailVerifyPostOkResponse`.
In the original code

```python
response = await api_options.recipe_implementation.verify_email_using_token(token, user_context)
    # we had to check is_ok since it notifies the http response from core which may contain an error in the status.
    if response.is_ok:
        # since we have not checked if the http response contains an error message
        # we need to validate if response.user is valid.  
        if response.user is None:
            
            raise Exception("Should never come here")
        return EmailVerifyPostOkResponse(response.user)
    return EmailVerifyPostInvalidTokenErrorResponse()
```
I've added `response.is_ok` validation now.
We can validate this in a more readable way 

```python
if response.is_ok:
    if response.user:
        return EmailVerifyPostOkResponse(response.user)
return EmailVerifyPostInvalidTokenErrorResponse()
```
If the `user` is `None`, we are anyways sending `EmailVerifyPostInvalidTokenErrorResponse`.

Please share your input.
Thanks.

Reply:
There was a little confusion initially.
Now it is resolved. updated this code.